alibi/explainers/pd_variance.py

```diff
@@ -0,0 +1,660 @@
+                ft_var = PartialDependenceVariance._compute_pd_variance_cat(pdv)
+            else:
+                ft_var = PartialDependenceVariance._compute_pd_variance_num(pdv)
+            # add extra dimension for later concatenation along the axis 0
+            feature_variance.append(ft_var[None, ...])  # type: ignore[index]
+
+        # stack the feature importance such that the array has the shape `F x T x N1 ... N(k-1)`
+        return np.concatenate(feature_variance, axis=0)
+
+    @staticmethod
+    def _compute_pd_variance_num(pd_values: np.ndarray) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for a numerical feature.
+
+        Parameters
+        ----------
+        pd_values
+            Array of partial dependence values for a numerical features of size `T x N1 x ... x Nk`, where `T` is the
+            number of targets and `Ni` is the number of feature values along the axis `i`.
+
+        Returns
+        -------
+        PD variance along the final axis for a numerical feature.
+        """
+        return np.std(pd_values, axis=-1, ddof=1, keepdims=False)
+
+    @staticmethod
+    def _compute_pd_variance_cat(pd_values: np.ndarray) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for a categorical feature.
+
+        Parameters
+        ----------
+        pd_values
+            Array of partial dependence values for a categorical feature of size `T x N1 x ... x Nk`, where `T` is the
+            number of targets and `Ni` is the number of feature values along the axis `i`.
+
+        Returns
+        -------
+        PD variance along the final axis for a categorical feature.
+        """
+        return (np.max(pd_values, axis=-1, keepdims=False) - np.min(pd_values, axis=-1, keepdims=False)) / 4
+
+    def _compute_feature_importance(self,
+                                    features: List[int],
+                                    pd_explanation: Explanation) -> Dict[str, Any]:
+        """
+        Computes the feature importance.
+
+        Parameters
+        ----------
+        features
+            List of features to compute the importance for.
+        pd_explanation
+            Partial dependence explanation object.
+
+        Returns
+        -------
+        Dictionary with all the keys necessary to build the explanation.
+        """
+        return {
+            'feature_deciles': pd_explanation.data['feature_deciles'],
+            'pd_values': pd_explanation.data['pd_values'],
+            'feature_values': pd_explanation.data['feature_values'],
+            'feature_names': pd_explanation.data['feature_names'],
+            'feature_importance': self._compute_pd_variance(features=features,
+                                                            pd_values=pd_explanation.data['pd_values']).T,
+        }
+
+    def _compute_feature_interaction(self,
```

Comment:
Remind me if this all works fine for mixed data, i.e. `(num, cat)`?

Reply:
Don't think I understand exactly what you mean. There shouldn't be any mixed data since the feature importance is defined for a single feature.

Reply:
This is feature interaction so 2 features? How does it work for mixed features?

Reply:
Sorry, my bad ... the feature interaction for mixed features works the same: fix one feature, let the other vary, and take the variance. If the fixed feature is numerical and the one that varies is categorical, then the standard deviation computation uses the range statistics. On the other hand, if the categorical if fixed and the numerical one varies, then the standard deviation computation uses the unbiased definition of the std ( divided by `N-1`). There shouldn't be any problem, and as yout can see, the method uses the internal `_compute_pd_variance` which is aware of the feature type.